### PR TITLE
Logic for the Booking component

### PR DESCRIPTION
**Pull request by dakatk**
- Functionality for the search fields on the booking page
- Top lodgings section
- Move search fields to a separate component

### CODE REVIEW

File: angular/src/app/modules/booking/featured-lodging/featured-lodging.component.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  setFeaturedLodgings(): void {
+    const lodgings: Lodging[] = new Array(6);
+
+    if (this.featuredLodgings) {
+      for (let index = 0; index < lodgings.length; index++) {
+        if (this.featuredLodgings[index].reviews) {
+          lodgings.push(this.featuredLodgings[index]);
+        }
+      }
+    }
```
Comment (escalonn) [inline angular/src/app/modules/booking/featured-lodging/featured-lodging.component.ts:25]: you'd use something like ngOnChanges hook for this sort of thing, not load event. and you're misusing one property here for both input and internal component stuff. make a 2nd property for featured lodgings, and rename the input one to what you're actually inputting, which is all lodgings

File: angular/src/app/modules/booking/search-bar/search-bar.component.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      }
+      return false;
+    }
+
+    // Most of this should be backend logic eventually,
+    // but this works for now
+    this.lodgings$.subscribe((lodgings) => {
+      const searchResults: Lodging[] = [];
+
+      lodgings.forEach((lodging) => {
```
Comment (escalonn) [inline angular/src/app/modules/booking/search-bar/search-bar.component.ts:104]: for..of is preferred to forEach generally

File: angular/src/app/modules/booking/booking/booking.component.html [hunk 3]
```diff
@@ -6,107 +6,14 @@ <h1 class="title is-2">booking</h1>
...
-                      accusamus nam nobis voluptas eius commodi officiis maxime quibusdam, aliquid
-                      distinctio, provident ut porro odio ducimus soluta et molestiae. Error, autem
-                      ipsa quis dignissimos commodi voluptas maxime quo.
-                    </p>
-                  </div>
-                </div>
-              </div>
-            </div>
-          </div>
+          <uic-spotlight [lodgings]="lodgings$ | async"></uic-spotlight>
```
[the remark this answers is not in the document]
Comment (dakatk) [reply angular/src/app/modules/booking/booking/booking.component.html:16]: yes

File: angular/src/app/modules/booking/featured-lodging/featured-lodging.component.ts (hunk 1 above)
Comment (escalonn) [reply angular/src/app/modules/booking/featured-lodging/featured-lodging.component.ts:19]: 2nd part of comment unaddressed

**Pull request merged**